coalib/parsing/DefaultArgParser.py

```diff
@@ -22,6 +23,23 @@ def _format_action_invocation(self, action):
             return ', '.join(parts)
 
 
+class PathArg(str):
```

Comment:
not sure if we should call it `path_arg` instead (thinking of `str`, `int`, etc because these are also all lowercase)

Reply:
and maybe we should just call it path? We could maybe use it then also as a settings type for bears (I hope this works....)

Reply:
@Makman2 I wouldn't make it a general setting type. I wouldn't allow `paths\with\backslashes` in config files. All your actual worries will come true then. And mine are the same :)

That's why the `PathArg` name, because it should really only be for the `ArgParser`, for cmdline args

Reply:
@Makman2 According to https://docs.python.org/3.6/distutils/setupscript.html#writing-the-setup-script ...

_"Note that any pathnames (files or directories) supplied in the setup script should be written using the Unix convention, i.e. slash-separated. The Distutils will take care of converting this platform-neutral representation into whatever is appropriate on your current platform before actually using the pathname. This makes your setup script portable across operating systems, which of course is one of the major goals of the Distutils. In this spirit, all pathnames in this document are slash-separated."_

**No** to backslashes in config file. **Yes** to backslashes on the Windows cmdline :)

Reply:
@Makman2 So an according bear setting type would again only make sense for the cmdline. When there will ever be a way to provide bear settings on the cmdline ;)

Reply:
@Makman2 Hmm... Actually I'm working on supporting bear settings as cmdline-like flags in `%%...Bear` cell magics as part of `bearsh`... So we should really think about some well-balanced typing mechanism there :)

It's like:

```python
In [*]: %%...Bear --setting value
   ...: code-to-anylize...
   ...: ...
```

Reply:
@Makman2 On the other hand, IPython always uses forward slashes in it's path auto-completion...

Reply:
But those should be maybe converted back to backslashes under Windows... Maybe not every external linter tool works with forward slashed paths on Windows... Hmm... The other way around :D We should later really think about such stuff...

Reply:
phew man you put many comments here ... 😅 

> I wouldn't make it a bear setting type. It would allow paths\with\backslashes in config files, right? All your actual worries will come true then. And mine are the same :)

Yeah that's a good thought, let's teach people the only way: The Linux way :P :P 
No but really for consistency this is a good point :+1:

Though we can consider reusing the type in a later issue :+1: